src/core/namespace.c

```diff
@@ -150,6 +154,14 @@ static const BindMount protect_system_strict_table[] = {
         { "/root",               READWRITE,    true  },      /* ProtectHome= */
 };
 
+/* If RootDirectory= is used, let's mount /sys and /proc into the it, but only as a fallback if the user hasn't mounted
+ * something there already. These mounts are hence overriden by any other explicitly configured mounts. */
+static const BindMount root_directory_table[] = {
+        { "/proc",               PROCFS,       false },
+        { "/dev",                BIND_DEV,     false },
+        { "/sys",                SYSFS,        false },
```

Comment:
Well, what about
https://github.com/systemd/systemd/pull/4395#issuecomment-258139966
> In order to mount sysfs in a user namespace, we need CAP_SYS_ADMIN rights
over the current net namespace (which happens implicitely when specifying --private-network entering a fresh new network ns with this cap set), otherwise it doesn't work because the kernel restricts the sysfs mount with -EPERM

?

Does this break the `RootDirectory=` inside the user namespace?